File: Java/src/Exists.java
import java.util.Arrays;

public class Exists {

    public static void main(String[] args) {
        int[] ints = {1, 2, 3, 4, 5};
        int k = 3;
        System.out.println(exists(ints, k));
    }

    static boolean exists(int[] ints, int k) {
        //Check if k exists in ints return true otherwise false
        //ints est trié par ordre croissant -> recherche binaire en O(log n)
        return Arrays.binarySearch(ints, k) >= 0;
    }
}

File: TS/exists.ts
function exists(ints: number[], k: number): boolean {
  // ints est trié par ordre croissant -> recherche binaire en O(log n)
  let low = 0;
  let high = ints.length - 1;

  while (low <= high) {
    const mid = (low + high) >>> 1;
    if (ints[mid] === k) {
      return true;
    }
    if (ints[mid] < k) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }

  return false;
}

export { exists };
